script.io.benchmark/default.py

```diff
@@ -0,0 +1,154 @@
+ #   Copyright (C) 2017 Lunatixz
+#
+#
+# This file is part of I/O Benchmark.
+#
+# I/O Benchmark is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# I/O Benchmark is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with I/O Benchmark.  If not, see <http://www.gnu.org/licenses/>.
+
+import os, sys, monkeytest
+import xbmc, xbmcgui, xbmcplugin, xbmcvfs, xbmcaddon
+
+
+# Plugin Info
+ADDON_ID       = 'script.io.benchmark'
+REAL_SETTINGS  = xbmcaddon.Addon(id=ADDON_ID)
+ADDON_NAME     = REAL_SETTINGS.getAddonInfo('name')
+ADDON_PATH     = (REAL_SETTINGS.getAddonInfo('path').decode('utf-8'))
+ADDON_VERSION  = REAL_SETTINGS.getAddonInfo('version')
+SETTINGS_LOC   = REAL_SETTINGS.getAddonInfo('profile')
+ICON           = os.path.join(ADDON_PATH, 'icon.png')
+FANART         = os.path.join(ADDON_PATH, 'fanart.jpg') 
+                        
+ADDON_FILE   = os.path.join((xbmc.translatePath(ADDON_PATH)),'test.tmp')
+SETTINGS_FILE= os.path.join((xbmc.translatePath(SETTINGS_LOC)),'test.tmp')
+
+if xbmcvfs.exists(SETTINGS_LOC) == False:
+    xbmcvfs.mkdirs(SETTINGS_LOC)
+     
+TEST_PATHS = None
+# change the constants below according to your needs
+WRITE_MB       = 128    # total MBs Wrote during test
+WRITE_BLOCK_KB = 1024   # KBs in each write block
+READ_BLOCK_B   = 512    # bytes in each read block (high values may lead to
+                        # invalid results because of system I/O scheduler        
+                        # file must be at drive under test
+  
+def repeat_to_length(string_to_expand, length):
+   return (string_to_expand * ((length/len(string_to_expand))+1))[:length]
+ 
+def log(msg, level = xbmc.LOGDEBUG):
+    xbmc.log(ADDON_ID + '-' + ADDON_VERSION + '-' + str(msg), level)
+    
+def showText(text, header=ADDON_NAME):
+    xbmcgui.Dialog().textviewer(header, text)
+     
+def okay(line1= '', line2= '', line3= '', header=ADDON_NAME):
+    xbmcgui.Dialog().ok(header, line1, line2, line3)
+   
+try:
```

Comment:
I don't know how much experience you have with python but just so you understand.

If source-code is free floating (not encapsulated within a method) it will be run whenever the file gets invoked.

What does this mean? In your case, if you or someone else would e.g. `import` this file, the whole source would be executed.
This is generally bad (unless you explicitly want this to happen).

In this case, we want the main routine to be run **only** when the script gets **executed**.
This is what `__name__` allows us to determine.